src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices
+{
+    public static class MarshalTests
+    {
+        [Fact]
+        public static void AllocHGlobal_Int32_Success()
+        {
+            IntPtr p = Marshal.AllocHGlobal(100);
+            Assert.NotEqual(IntPtr.Zero, p);
+            Marshal.FreeHGlobal(p);
+        }
+
+        [Fact]
+        public static void AllocHGlobal_IntPtr_Success()
+        {
+            IntPtr p = Marshal.AllocHGlobal((IntPtr)100);
+            Assert.NotEqual(IntPtr.Zero, p);
+            Marshal.FreeHGlobal(p);
+        }
+
+        [Fact]
+        public static void AllocHGlobal_OOM()
+        {
+            if (IntPtr.Size == 8)
```

Comment:
This only tests the 64-bit runtime. Obviously, you intended that. Do we also test the 32-bit runtime `IntPtr.Size == 4` ?